app/javascript/src/NewService/components/FormElements/FormWrapper.jsx

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+import React from 'react'
+import type {FormProps} from 'NewService/types'
+
+const FormWrapper = (props: FormProps) => {
+  const {id, formActionPath, CSRFToken, HiddenServiceDiscoveryInput, submitText, ListItems, ListItemsProps} = props
+  return (
+    <form className='formtastic service'
+      id={id} action={formActionPath}
+      acceptCharset="UTF-8" method="post"
+    >
+      <input name="utf8" type="hidden" value="✓"/>
+      <CSRFToken />
+      {HiddenServiceDiscoveryInput ? <HiddenServiceDiscoveryInput/> : null}
+      <fieldset className="inputs" name="Service">
+        <legend><span>Service</span></legend>
+        <ol>
+          {ListItems ? <ListItems {...ListItemsProps}/> : null}
```

Comment:
What it troubles me is the fact that we might not have that component... but we could have its props? This component knows way too much. The `ListItems` props should be managed just inside `ListItems`. Find a way to pass the component ready to render or not at all. Or with a default state that would replace `null`